package/MDAnalysis/coordinates/chain.py

```diff
@@ -155,8 +242,84 @@ def __init__(self, filenames, **kwargs):
         #: source for trajectories frame (fakes trajectory)
         self.__chained_trajectories_iter = None
 
+        # calculate new start_frames to have a time continuous trajectory.
+        if continuous:
+            filetypes = np.unique([r.format for r in self.readers])
+            if not len(filetypes) == 1:
+                raise RuntimeError("Continuous only supported with all files "
+                                   "are from the same format. found {}".format(filetypes))
+            if np.any(np.array(n_frames) == 1):
+                raise RuntimeError("Need at least two frames in every trajectory")
+            if filetypes[0] not in ['XTC', 'TRR']:
+                raise RuntimeError("continuous only supported for xtc and trr format")
+
+            # TODO: allow floating point precision in dt check
+            dt = self._get_same('dt')
+            n_frames = np.asarray(self._get('n_frames'))
+            self.dts = np.ones(self.dts.shape) * dt
+
+            # the sorting needs to happen on two levels. The first major level
+            # is by start times and the second is by end times.
+            # The second level of sorting is needed for cases like:
+            # [0 1 2 3 4 5 6 7 8 9] [0 1 2 4]
+            # to
+            # [0 1 2 4] [0 1 2 3 4 5 6 7 8 9]
+            # after that sort the chain reader will work
+            times = []
+            for r in self.readers:
+                r[0]
+                start = r.ts.time
+                r[-1]
+                end = r.ts.time
+                times.append((start, end))
+            # sort step
+            sort_idx = multi_level_argsort(times)
+            self.readers = [self.readers[i] for i in sort_idx]
+            self.filenames = self.filenames[sort_idx]
+            self.total_times = self.dts * n_frames[sort_idx]
+
+            # filter step: remove indices if we have complete overlap
+            if len(self.readers) > 1:
+                used_idx = filter_times(np.array(times)[sort_idx], dt)
+
+                self.readers = [self.readers[i] for i in used_idx]
+                self.filenames = self.filenames[used_idx]
+                self.total_times = self.dts[used_idx] * n_frames[used_idx]
+
+            # rebuild lookup table
+            sf = [0, ]
+            n_frames = 0
+            print(len(list(zip(self.readers[:-1], self.readers[1:]))))
+            print(len(self.readers[:-1]))
+            for r1, r2 in zip(self.readers[:-1], self.readers[1:]):
+                r2[0], r1[0]
+                start_time = r2.time
+
+                # trajectory doesn't contain valid frames
+                if r1.time >= start_time:
+                    # this ensure bisect works correctly
+                    sf.append(sf[-1])
+                    continue
+
+                print(start_time, n_frames, dt)
+                if start_time > n_frames * dt:
+                    warnings.warn("Missing frame in continuous chain", UserWarning)
```

Comment:
logger.warn ?

Reply:
I usually prefer this. It's more verbose and will be noticed. I personally never turn on the logging and never see any of the logging messages.